tests/testCountInputs.py

```diff
@@ -0,0 +1,207 @@
+import lsst.afw.image as afwImage
+import lsst.afw.coord as afwCoord
+import lsst.afw.detection as afwDetection
+import lsst.meas.base as measBase
+from lsst.afw.geom.polygon import Polygon
+
+def ccdVennDiagram(exp, showImage=True, legendLocation='best'):
+    '''
+    Create a figure with the bounding boxes for each of the images which go into a coadd,
+    over-plotting the given exposure object.
+    exp [in] - (Exposure) The exposure object to plot, must be the product of a coadd
+    Optional:
+    showImage [in] - (Bool) Plot the data in exp in addition to just it's bounding box, default True
+    LegendLocation [in] - (String) Matplotlib legend location code, can be: 'best', 'upper right',
+                             'upper left', 'lower left', 'lower right', 'right', center left',
+                             'center right', 'lower center', 'upper center', 'center'
+    '''
+    # Create the figure object
+    fig = plt.figure()
+    # Use all the built in matplotib line style attributes to create a list of the possible styles
+    linestyles = ['solid','dashed','dashdot','dotted']
+    colors = ['b','g','r','c','m','y','k']
+    # Calculate the cartisian product of the styles, and randomize the order, to help each CCD get
+    # it's own color
+    pcomb = np.random.permutation(list(itertools.product(colors,linestyles)))
+    # Filter out a black solid box, as that will be the style of the given exp object
+    pcomb = pcomb[((pcomb[:,0] == 'k') * (pcomb[:,1] == 'solid')) == False]
+    # Get the image properties
+    origin = lsst.afw.geom.PointD(exp.getXY0())
+    mainBox = exp.getBBox().getCorners()
+    # Plot the exposure
+    plt.gca().add_patch(patches.Rectangle((0,0),*list(mainBox[2]-mainBox[0]), fill=False, label="exposure"))
+    # Grab all of the CCDs that went into creating the exposure
+    ccds = exp.getInfo().getCoaddInputs().ccds
+    # Loop over and plot the extents of each ccd
+    for i,ccd in enumerate(ccds):
+        ccdBox = lsst.afw.geom.Box2D(ccd.getBBox())
+        ccdCorners = ccdBox.getCorners()
+        coaddCorners = [exp.getWcs().skyToPixel(ccd.getWcs().pixelToSky(point)) + \
+                        (lsst.afw.geom.PointD() - origin) for point in ccdCorners]
+        plt.gca().add_patch(patches.Rectangle(coaddCorners[0], *list(coaddCorners[2]-coaddCorners[0]),
+                            fill=False, color=pcomb[i][0], ls=pcomb[i][1], label="CCD{}".format(i)))
+    # If showImage is true, plot the data contained in exp as well as the boundrys
+    if showImage:
+        plt.imshow(exp.getMaskedImage().getArrays()[0], cmap='Greys', origin='lower')
+        plt.colorbar()
+    # Adjust plot parameters and plot
+    plt.gca().relim()
+    plt.gca().autoscale_view()
+    ylim = plt.gca().get_ylim()
+    xlim = plt.gca().get_xlim()
+    plt.gca().set_ylim(1.5*ylim[0], 1.5*ylim[1])
+    plt.gca().set_xlim(1.5*xlim[0], 1.5*xlim[1])
+    plt.legend(loc=legendLocation)
+    fig.canvas.draw()
```

Comment:
Don't you need to do something with the figure after you've generated it? `plt.show()` or save it to a file or similar?